src/lib/components/private/QIconSnippet.svelte

```diff
@@ -0,0 +1,28 @@
+<!-- This component should be used when an icon can be set from props. The icon can either be undefined, a string or a Snippet so this componnet handles it all. -->
+<script lang="ts">
+  import QIcon from "$components/icon/QIcon.svelte";
+  import type { QIconProps } from "$components/icon/props";
+  import type { Snippet } from "svelte";
+  import type { MaterialSymbol } from "material-symbols";
+
+  interface IconSnippetProps extends Omit<QIconProps, "name"> {
+    icon?: MaterialSymbol | Snippet;
+    defaultIcon?: MaterialSymbol | Snippet;
+  }
+
+  let { icon, defaultIcon, ...props }: IconSnippetProps = $props();
+</script>
+
+{#if icon}
```

Comment:
Can't we have a `$derived` that is `icon ?? defaultIcon`? Would save markup.

Reply:
8d94ffec09f22cffa5a14a5a91fb732894e765fd